### ISSUE TITLE
Create Playerclass Wiki Documentation On the Current State of PlayerClass and rooms for future improvement

### ISSUE THREAD

**Issue by Vincent-Zhang4135** [uchicago-cs/chiventure#1680]
As we are wrapping up the last sprint of our team, we want to make a document that records the current state of PlayerClasses so that there is less software archaeology for future devs. Coming onto this feature this year, many of us were a little confused even with the quite extensive documentation, and so we thought it would be helpful to create documentation that is more tailored towards where there can be room of improvement so

The wiki document link can be found here: [Current State of PlayerClass](https://github.com/uchicago-cs/chiventure/wiki/Player-Class=~-Current-State-of-Player-Class)

**Comment by katie-gao** [comment]
I added skills to this. For the current state, we have some skill tree functionalities, where each of the playerclasses (see playerclass list) have damages and skills. There is also functionality to add more skills. There is still room for improvement in the skills, especially multitasking and adding skills for the new npc-specific prefabs.

**Comment by jroven** [comment]
I added NPC integration to this page. The current state includes the new NPC-specific prefab classes, along with the indefinite movement multiplier feature. I linked the [Player Class NPC Integration](https://github.com/uchicago-cs/chiventure/wiki/Player-Class-~-Player-Class-NPC-Integration) wiki page for the future dev section since there was a lot on it that we did not get around to implementing this quarter.

**Comment by Vincent-Zhang4135** [comment]
I added the Battles integration to this page. This includes the description of the integration we made on consuming battle_item consumables to change player stats, but also the additional areas of improvements in further integration battle items and equipments. I also added a description of the multi class functionality, which is the main thing that we will be presenting in our demo.

**Comment by katie-gao** [comment]
I am adding multiclass integration to this page. This will include the prefabs we have, the functionalities we have with multiclasses, and where we can improve. It is also similar to https://github.com/uchicago-cs/chiventure/wiki/Player-Class-~-Multiclassing:-Design multiclass design.

**Comment by katie-gao** [comment]
In this issue, we added a page on the wiki page called Current State of Player Class, to write out all the things we worked on this past year, and directions for next year. While there were user stories, it was not always clear cut what we could work on and what was possible to work on. In particular, we covered multiclass, skills, npc integration, and battle integration. As a team, these were the four major ideas that we made significant progress on. For multiclassing, it is fully functional now with no seg faults, and there are four new prefab multiclasses )"hexblade", "infernal", "paladin", "enchanter", and "assassin") that will be displayed on the demo. Second, skills made huge progress, with every class having combat ability skills, and scaled damages. Third, we npc movement that scales was implemented, and there was a creation of npc specific prefabs. Lastly, there was custom battle item integration alongside playerclass, where more damage was done with a particular battle item and playerclass that matched. 

Then, we gave suggestions on improving the four topics we worked on. In general, there needs to be more testing. A lot of the functions right now work for the most part, but when working on the demo, it is easy to encounter seg faults or just have the functions not working properly. There is still work that needs to be done with npc and skillstree teams, and the npc specific prefabs should have skills. There can be more battle item integrations, especially with offensive items. For now, this is a good summary of the work we had done and the potential playerclass still has. It also will help with the writing of the final report.

In particular, I worked on multiclass and skill sections of the page, vincent worked on battle integrations, and jonathan worked on npc integration. These are the major topics that we covered - so we are closing the issue.
